src/Avalonia.Visuals/Rect.cs

```diff
@@ -421,11 +421,48 @@ public Rect Translate(Vector offset)
         }
 
         /// <summary>
-        /// Gets the union of two rectangles.
-        /// </summary>
-        /// <param name="rect">The other rectangle.</param>
-        /// <returns>The union.</returns>
-        public Rect Union(Rect rect)
+		/// Normalizes the rectangle so both the <see cref="Width"/> and <see cref="Height"/> are positive, without changing the location of the rectangle
+		/// </summary>
+        /// <returns>Normalized Rect</returns>
+		/// <remarks>
+		/// Empty rect will be return when Rect contains invalid values. Like NaN.
+		/// </remarks>
+		public Rect Normalize()
+        {
+            Rect rect = this;            
+
+            if(double.IsNaN(rect.Right) || double.IsNaN(rect.Bottom) || double.IsNaN(rect.X) || double.IsNaN(rect.Y) || double.IsNaN(Height) || double.IsNaN(Width))
+            {
+                return Rect.Empty;
+            }
+
+            if (rect.Width < 0)
+            {
+                var old = X;
+                var x = X + Width;
+                var width = old - x;
+
+                rect = rect.WithX(x).WithWidth(width);
```

Comment:
Can we use the ctor here? I know there is not much to gain here.

Reply:
i think this would be optimized out anyway and its more readable like this.